### DISCUSSION TITLE
Browser Support

### DISCUSSION THREAD

**Discussion by mchatrvd** [PostgREST/postgrest#3668] [Q&A]
I understand that there is a curl command option to invoke PostgREST API service. Is there any option to invoke service from the browser itself ?

**Comment by laurenceisla** [comment]
Well, the browser does a `GET` request when you type the URL in the browser's Address Bar. But since PostgREST provides an API, it's better to work with it using JavaScript, [for example, via fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch).

Once you're familiarized with it, you may check libraries like [`postgrest-js`](https://github.com/supabase/postgrest-js), or [`vue-postgrest`](https://github.com/technowledgy/vue-postgrest) for the Vue.js JavaScript framework.

**Reply by develop7** [reply]
@laurenceisla we could bundle something like https://github.com/swagger-api/swagger-ui to postgrest, it's a documentation browser that allows the user to issue API requests on the spot

**Reply by wolfgangwalther** [reply]
There is no need to bundle that. It's already mentioned in the install section of the docs how to set it up with docker compose btw: https://docs.postgrest.org/en/v12/explanations/install.html#containerized-postgrest-and-db-with-docker-compose

**Reply by laurenceisla** [reply]
> it's a documentation browser that allows the user to issue API requests on the spot

Ah, right, other option to make API requests from the browser is by using a tool like SwaggerUI. I (mis)understood that they wanted to use it in their web app.

> There is no need to bundle that

This, the example given in the docs already works perfectly.

**Reply by steve-chavez** [reply]
> we could bundle something like https://github.com/swagger-api/swagger-ui to postgrest

Since we have https://postgrest.org/en/v12/references/api/media_type_handlers.html maybe we could bundle swagger-ui. I'm guessing it's just an HTML page plus some javascripts, which could be contained in db functions returning `text/html` and `application/javascript`. Possible to have a built-in/overridable page too.

docker compose is an option too but not as convenient as a builtin dashboard.

**Reply by wolfgangwalther** [reply]
> Since we have https://postgrest.org/en/v12/references/api/media_type_handlers.html maybe we could bundle swagger-ui. I'm guessing it's just an HTML page plus some javascripts, which could be contained in db functions returning `text/html` and `application/javascript`. Possible to have a built-in/overridable page too.

We should not bundle that into core. This could be something provided in "contrib", though. Request the root endpoint as html? Swagger-UI. Request it as json? OpenAPI output.
